jargon-core/src/main/java/org/irods/jargon/core/packinstr/StartupPack.java

```
@@ -2,19 +2,34 @@
 
 import org.irods.jargon.core.connection.IRODSAccount;
 import org.irods.jargon.core.exception.JargonException;
+// import org.irods.jargon.core.connection.IRODSSession;
 
 public class StartupPack extends AbstractIRODSPackingInstruction {
 
 	public static final String PI_TAG = "StartupPack_PI";
 	public static final String CHALLENGE = "challenge";
-	public static final String SP_OPTION_IINIT = "iinit";
 	public static final String protocolType = "1"; // 1 = xml protocol
 	private final IRODSAccount irodsAccount;
 	private int reconnFlag = 0;
 	private int connectCnt = 0;
-	private String option = "";
+	// default name when reporting through sp_option/spOption; allows jargon
+	// applications to report with a name through ips
+	private static final String DEFAULT_APPLICATION_NAME = "jargon";
+	private static String option = DEFAULT_APPLICATION_NAME;
 	public static final String NEGOTIATE_OPTION = "request_server_negotiation";
 
+	public static String getApplicationName() {
+		return option;
+	}
+
+	public static void setApplicationName(String name) {
+		if (name == null) {
```

Should an empty string be allowed?

Hmm ...

_This isn't a request to change anything, yet._

---

Hmmm, I don't see how an empty name would be helpful when running `ips`. Maybe if they purposefully set an empty string, we can use "UNKNOWN"? Hmmmm, but then they could also just set "UNKNOWN" themselves.

not really sure about this one

---

Let's throw an exception if someone passes null or an empty string to this function.

We can assume if they are calling this, they want to set a non-empty name. Doing anything else doesn't make sense.

---

Sounds great. Changed it so that if `name == null || name.trim().isEmpty()` (the trim().isEmpty() means no strings with just spaces in it), we're throwing an `IllegalArgumentException`. 

If you want to set the name to "jargon" or something like that when a null or empty name is passed in, just let me know.